File: pact-consumer/src/main/java/com/example/SuppliesAnalyser.java
package com.example;

import java.util.List;

public class SuppliesAnalyser {

    public double countAverageItemWeight(List<Supply> supplies) {
        int sumWeights = supplies.stream()
                .mapToInt(Supply::getTotalWeight)
                .sum();
        int totalItems = supplies.stream()
                .mapToInt(Supply::getCount)
                .sum();
        return (double) sumWeights / totalItems;
    }
}

File: spring-consumer/src/main/kotlin/com/example/SuppliesAnalyser.kt
package com.example

class SuppliesAnalyser {

    fun countAverageItemWeight(supplies: List<Supply>): Double {
        val sumWeights = supplies.map { it.totalWeight }.sum()
        val totalItems = supplies.map { it.count }.sum()
        return sumWeights.toDouble() / totalItems
    }
}
